src/sentry/api/serializers/models/group.py

```diff
@@ -900,22 +900,32 @@ def __init__(
         if end_params:
             self.end = min(end_params)
 
-        self.conditions = (
-            [
-                convert_search_filter_to_snuba_query(
-                    search_filter,
-                    params={
-                        "organization_id": organization_id,
-                        "project_id": project_ids,
-                        "environment_id": environment_ids,
-                    },
-                )
-                for search_filter in search_filters
-                if search_filter.key.name not in self.skip_snuba_fields
-            ]
-            if search_filters is not None
-            else []
-        )
+        conditions = []
+        if search_filters is not None:
+            for search_filter in search_filters:
+                if search_filter.key.name not in self.skip_snuba_fields:
+                    formatted_conditions, projects_to_filter, group_ids = format_search_filter(
+                        # convert_search_filter_to_snuba_query(
+                        search_filter,
+                        params={
+                            "organization_id": organization_id,
+                            "project_id": project_ids,
+                            "environment_id": environment_ids,
+                        },
+                    )
+                    conditions.append(
+                        formatted_conditions[0]
+                        if formatted_conditions
+                        else convert_search_filter_to_snuba_query(
+                            search_filter,
+                            params={
+                                "organization_id": organization_id,
+                                "project_id": project_ids,
+                                "environment_id": environment_ids,
+                            },
+                        )
+                    )
+        self.conditions = conditions
```

Comment:
I wonder if its worthwhile to filter out conditions like `[None, "=", "value"]` that couldn't resolve the search key alias, or at least throw an error that bubbles up to the front-end.

Reply:
I mean it just says "Request failed" basically. It's not actionable from a user perspective and we are logging + creating an issue from it. Feel like we can leave as-is

Reply:
True. Maybe we could throw a specific error when we transform the `SearchFilter` -> `Condition` and the resulting condition has a `None` key.name. That way the user can fix their query instead of just getting a request failed and not know what's going on?

Reply:
eh i mean it's not really user actionable because we're converting the name to `None`. it should be that we throw the error like now IMO

Reply:
Currently, when searching with a query like `issue:DOESNT-EXIST` the front-end shows an internal error. 
![Screen Shot 2023-02-02 at 3 07 37 PM](https://user-images.githubusercontent.com/101606877/216471417-3387e723-4152-4306-8163-21d8af20a18c.png)

Your PR should fix the cases for issue and project querying but any future keys that don't transform from SearchFilter -> Condition will result in a non-helpful generic Internal error. If we do a check after the search filter transformation for any conditions that have a `None` key, we could explicitly raise an error that would show up in the front-end so at least the user has a better idea why the query failed.

Reply:
Right I'm just saying that this really is an internal error and there's nothing the user can do about it. I don't think we really want to add a message like "we have a bug"

Reply:
Yeah, fair enough.